homework_4/01_bets_api/internal/api/controllers/controller.go

```diff
@@ -0,0 +1,69 @@
+package controllers
+
+import (
+	"github.com/superbet-group/code-cadets-2021/homework_4/01_bets_api/internal/infrastructure/sqlite"
+	"log"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+)
+
+// Controller implements handlers for web server requests.
+type Controller struct {
+	dbService  DbService
+	dbExecutor sqlite.DatabaseExecutor
+}
+
+// NewController creates a new instance of Controller
+func NewController(dbService DbService, dbExecutor sqlite.DatabaseExecutor) *Controller {
+	return &Controller{
+		dbService:  dbService,
+		dbExecutor: dbExecutor,
+	}
+}
+
+// GetBetByID handlers gets the bet with id.
+func (e *Controller) GetBetByID() gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		id := ctx.Param("id")
+
+		betResponse, err := e.dbService.GetBetByID(id, e.dbExecutor) //get id from uri
+		if err != nil {
+			log.Println(err)
+			ctx.String(http.StatusNotFound, "no bet for this id")
+			return
+		}
+
+		ctx.JSON(http.StatusOK, betResponse)
```

Comment:
you need to map to the DTO model before you return the value.
Controller should have the logic for DTO mapping

Reply:
The same should apply to all of the controller functions